src/WindowsPackageManager/ConfigurationStaticFunctions.cpp

```diff
@@ -81,11 +103,78 @@ namespace ConfigurationShim
 
         winrt::Microsoft::Management::Configuration::ConfigurationProcessor CreateConfigurationProcessor(winrt::Microsoft::Management::Configuration::IConfigurationSetProcessorFactory const& factory)
         {
+            if (!m_statics)
+            {
+                THROW_HR(APPINSTALLER_CLI_ERROR_PACKAGE_IS_STUB);
+            }
+
             auto result = m_statics.CreateConfigurationProcessor(factory);
             result.as<AppInstaller::WinRT::ILifetimeWatcher>()->SetLifetimeWatcher(CreateLifetimeWatcher());
             return result;
         }
 
+        bool IsConfigurationEnabled()
+        {
+            return !IsStubPackage();
+        }
+
+        winrt::Windows::Foundation::IAsyncActionWithProgress<uint32_t> EnableConfigurationAsync()
+        {
+            if (IsConfigurationEnabled())
+            {
+                return;
+            }
+
+            co_await winrt::resume_background();
+
+            SetStubPreferred(false);
+
+            PackageManager packageManager;

```

Comment:
Does this actually create the object properly?  I remember seeing newer version of CppWinRT do handle that, but I don't think we have updated to them?

My gut would be that we need to create them with `make_self`.

Reply:
Talked offline